lib/auth/init.go

```diff
@@ -358,35 +357,31 @@ func Init(cfg InitConfig, opts ...ServerOption) (*Server, error) {
 				Namespace: apidefaults.Namespace,
 			},
 			Spec: types.CertAuthoritySpecV2{
-				ClusterName:  cfg.ClusterName.GetClusterName(),
-				Type:         types.UserCA,
-				SigningKeys:  [][]byte{priv},
-				SigningAlg:   sshutils.ParseSigningAlg(sigAlg),
-				CheckingKeys: [][]byte{pub},
-				TLSKeyPairs:  []types.TLSKeyPair{{Cert: certPEM, Key: keyPEM}},
+				ClusterName: cfg.ClusterName.GetClusterName(),
+				Type:        types.UserCA,
+				ActiveKeys: types.CAKeySet{
+					SSH: []*types.SSHKeyPair{{
+						PrivateKey:     priv,
+						PrivateKeyType: types.PrivateKeyType_RAW,
+						PublicKey:      pub,
+					}},
+					TLS: []*types.TLSKeyPair{{
+						Cert:    certPEM,
+						Key:     keyPEM,
+						KeyType: types.PrivateKeyType_RAW,
+					}},
+				},
+				SigningAlg: sshutils.ParseSigningAlg(sigAlg),
 			},
 		}
 
-		if err := asrv.Trust.UpsertCertAuthority(userCA); err != nil {
-			return nil, trace.Wrap(err)
-		}
-	} else if len(userCA.GetTLSKeyPairs()) == 0 {
-		log.Infof("Migrate: generating TLS CA for existing user CA.")
```

Comment:
for host and user CA, by this point all CAs should have TLS keypairs provisioned
we can remove this migration